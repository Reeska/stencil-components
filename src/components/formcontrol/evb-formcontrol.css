:host {
  align-items: center;
  display: flex;
  flex-direction: row;
  margin: .5rem 0;
}

:host(.vertical) {
  align-items: flex-start;
  flex-direction: column;
}

::slotted(input), ::slotted(textarea), ::slotted(select), ::slotted(option){
  border: 1px solid var(--evb-border-light);
  flex: 2 1 auto;
  font-size: 1em;
  line-height: normal;
  padding: calc(.5rem - 1px);
  width: calc(100% - 1rem);
}

::slotted(evb-range) {
  padding: .5rem 0;
}

::slotted(input:focus), ::slotted(textarea:focus) {
  border-color: var(--evb-primary-color);
  outline: 0;
}

::slotted(::placeholder) {
  color: var(--evb-border-light);
}

::slotted(label) {
  margin: .5rem;
  margin-left: 0;
  max-width: 25%;
  width: 150px;
}
