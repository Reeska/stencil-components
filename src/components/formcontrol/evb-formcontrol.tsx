import { Component, ComponentInterface, Element, h, Prop } from '@stencil/core';

@Component({
  tag: 'evb-formcontrol',
  shadow: true,
  styleUrls: [
    'evb-formcontrol.css',
    'evb-formcontrol-textarea.css'
  ]
})
export class EvbFormControl implements ComponentInterface {

  @Element() host: HTMLElement;
  /**
   * renders childs (eg. label + input) on the same line
   */
  @Prop() vertical = false;

  render() {
    this.host.classList.toggle('vertical', this.vertical);
    return (
      <slot />
    );
  }
}
