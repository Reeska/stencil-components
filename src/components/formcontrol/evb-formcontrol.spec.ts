import { EvbFormControl } from "./evb-formcontrol";

describe('EvbFormControl', () => {
  let component: EvbFormControl;

  beforeEach(() => component = new EvbFormControl());

  it('should render', () => {
    expect(component.render()).toMatchSnapshot();
  });
});
