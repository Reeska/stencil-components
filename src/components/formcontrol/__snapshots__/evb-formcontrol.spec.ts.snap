// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvbFormControl should render 1`] = `
Object {
  "elm": undefined,
  "ishost": false,
  "vattrs": null,
  "vchildren": null,
  "vkey": undefined,
  "vname": undefined,
  "vtag": "slot",
  "vtext": undefined,
}
`;
